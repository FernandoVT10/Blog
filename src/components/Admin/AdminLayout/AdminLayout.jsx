import Layout from "../../Layout";
import AdminNavbar from "../AdminNavbar";

import "./AdminLayout.scss";

export default ({ children, title }) => {
    return (
        <Layout title={title}>
            <div className="admin-layout">
                <AdminNavbar />

                <div className="container-fluid admin-layout__wrapper">
                    { children }
                </div>
            </div>
        </Layout>
    );
};