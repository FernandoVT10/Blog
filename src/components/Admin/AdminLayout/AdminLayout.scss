.admin-layout {
    display: flex;
    overflow: hidden;
    padding-bottom: 20px;

    &__loader {
        height: 100vh;
        display: flex;
        justify-content: center;
        align-items: center;
        color: var(--secondary);
        background: var(--black);

        & span {
            width: 80px;
            height: 80px;
        }
    }

    &__wrapper {
        position: relative;
        margin-top: 30px;
        left: 25px;
        width: calc(100% - 50px);
    }

    @media screen and (max-width: 768px) {
        padding-bottom: 70px;

        &__wrapper {
            left: 0;
            width: 100%;
        }
    }
}