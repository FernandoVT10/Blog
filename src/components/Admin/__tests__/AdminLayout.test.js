import { create } from "react-test-renderer";
import AdminLayout from "../AdminLayout/AdminLayout";

test("Check if <AdminLayout/> renders correctly", () => {
    const component = create(<AdminLayout/>);

    expect(component.toJSON()).toMatchSnapshot();
});