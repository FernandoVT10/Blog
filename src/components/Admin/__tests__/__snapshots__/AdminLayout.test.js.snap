// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if <AdminLayout/> renders correctly 1`] = `
<div>
  <div
    className="admin-layout"
  >
    <nav
      className="admin-navbar "
    >
      <ul
        className="admin-navbar__items-container"
      >
        <li
          className="admin-navbar__item-header"
        >
          <span
            className="admin-navbar__item-icon"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fas fa-bars"
            />
          </span>
          <h4
            className="admin-navbar__item-title"
          >
            Admin Blog
          </h4>
        </li>
        <li
          className="admin-navbar__item"
        >
          <a
            href="/"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="admin-navbar__item-icon"
            >
              <i
                aria-hidden="true"
                className="fas fa-globe-americas"
              />
            </span>
            <span
              className="admin-navbar__item-text"
            >
              Back to Website
            </span>
          </a>
        </li>
        <li
          className="admin-navbar__item"
        >
          <a
            href="/admin"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="admin-navbar__item-icon"
            >
              <i
                aria-hidden="true"
                className="fas fa-th-large"
              />
            </span>
            <span
              className="admin-navbar__item-text"
            >
              Dashboard
            </span>
          </a>
        </li>
        <li
          className="admin-navbar__item"
        >
          <a
            href="/admin/articles"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="admin-navbar__item-icon"
            >
              <i
                aria-hidden="true"
                className="fas fa-images"
              />
            </span>
            <span
              className="admin-navbar__item-text"
            >
              Articles
            </span>
          </a>
        </li>
        <li
          className="admin-navbar__item"
        >
          <a
            href="/admin/projects"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="admin-navbar__item-icon"
            >
              <i
                aria-hidden="true"
                className="fas fa-folder-open"
              />
            </span>
            <span
              className="admin-navbar__item-text"
            >
              Projects
            </span>
          </a>
        </li>
        <li
          className="admin-navbar__item"
        >
          <a
            href="/admin/messages"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            <span
              className="admin-navbar__item-icon"
            >
              <i
                aria-hidden="true"
                className="fas fa-envelope"
              />
              <span
                className="admin-navbar__item-badge"
              >
                0
              </span>
            </span>
            <span
              className="admin-navbar__item-text"
            >
              Messages
            </span>
          </a>
        </li>
      </ul>
    </nav>
    <div
      className="container-fluid admin-layout__wrapper"
    />
  </div>
  <script
    src="https://cdn.jsdelivr.net/npm/chart.js@2.9.3/dist/Chart.min.js"
  />
  <script
    src="https://code.jquery.com/jquery-3.4.1.slim.min.js"
  />
  <script
    src="https://cdn.jsdelivr.net/npm/popper.js@1.16.0/dist/umd/popper.min.js"
  />
  <script
    src="https://stackpath.bootstrapcdn.com/bootstrap/4.4.1/js/bootstrap.min.js"
  />
  <script
    src="https://kit.fontawesome.com/63ef8f1397.js"
  />
</div>
`;
